java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -75,57 +76,28 @@ public RESTApiSanitizerImpl(@Assisted RESTRules rules,
     ReversibleTokenizationStrategy reversibleTokenizationStrategy;
     @Inject
     UrlSafeTokenPseudonymEncoder urlSafePseudonymEncoder;
-
-
-
     @Inject
     JsonSchemaFilterUtils jsonSchemaFilterUtils;
 
-    Map<Endpoint, Pattern> getCompiledAllowedEndpoints() {
-        if (compiledAllowedEndpoints == null) {
-            synchronized ($writeLock) {
-                compiledAllowedEndpoints = rules.getEndpoints().stream()
-                     .collect(Collectors.toMap(Function.identity(),
-                           endpoint -> Pattern.compile(effectiveRegex(endpoint), CASE_INSENSITIVE)));
-            }
-        }
-        return compiledAllowedEndpoints;
-    }
-
-    @VisibleForTesting
-    String effectiveRegex(Endpoint endpoint) {
-        return Optional.ofNullable(endpoint.getPathRegex())
-            .orElseGet(() -> "^" + endpoint.getPathTemplate().replaceAll("\\{.*?\\}", "[^/]+") + "$");
-    }
-
-    JsonSchemaFilterUtils.JsonSchemaFilter getRootDefinitions() {
-        if (rootDefinitions == null) {
-            synchronized ($writeLock) {
-              if (rootDefinitions == null) {
-                      rootDefinitions = JsonSchemaFilterUtils.JsonSchemaFilter.builder().definitions(rules.getDefinitions()).build();
-              }
-            }
-        }
-        return rootDefinitions;
-    }
 
 
 
     @Override
     public boolean isAllowed(@NonNull String httpMethod, @NonNull URL url) {
         String relativeUrl = URLUtils.relativeURL(url);
 
+
         if (rules.getAllowAllEndpoints()) {
             return true;
         } else {
             return getCompiledAllowedEndpoints().entrySet().stream()
-                .filter(entry -> entry.getValue().matcher(relativeUrl).matches())
+                .filter(entry -> entry.getKey().getPathRegex() != null && entry.getValue().matcher(relativeUrl).matches()
+                    || (entry.getKey().getPathTemplate() != null && entry.getValue().matcher(url.getPath()).matches() && allowedQueryParams(entry.getKey(), URLUtils.queryParamNames(url))))
                 .filter(entry -> entry.getKey().getAllowedMethods()
                     .map(methods -> methods.stream().map(String::toUpperCase).collect(Collectors.toList())
                             .contains(httpMethod.toUpperCase()))
                     .orElse(true))
-                .filter(entry -> entry.getKey().getAllowedQueryParamsOptional()
-                    .map(allowedParams -> allowedParams.containsAll(URLUtils.queryParamNames(url))).orElse(true))
+                .filter(entry -> allowedQueryParams(entry.getKey(), URLUtils.queryParamNames(url))) // redundant in the pathTemplate case
```

Comment:
why don't you move it out from the L95 if it's going to be done here either way?